### PR TITLE
*: Make DDL unaffected by killing TiDB instance

### PR DESCRIPTION
**Pull request by zimulala**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/43854

Problem Summary:
1. This [issue](https://github.com/pingcap/tidb/pull/35803) support canceling DDL statements with KILL（related PR:https://github.com/pingcap/tidb/pull/35923）.
---**The original PR that supported this didn't seem to say cancel DDL on "kill TiDB"**

2. But after these PRs, when we do `kill TiDB`, this TiDB's DDLs **may be** canceled(In some cases it can't be cancelled).
--- **Cancel DDL is not fully supported in the case of "kill TiDB".**

3. This [issue](https://github.com/pingcap/tidb/issues/39751) need do "Pause user DDL when upgrading TiDB", but these DDLs may be canceled by `kill TiDB`. So we consider `kill TiDB` can not cancel DDL.

4. We'll deal with this in detail when we distinguish between asynchronous and synchronous DDLs.

### What is changed and how it works?
If don't we kill TiDB can't cancel DDLs.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [x] Breaking backward compatibility

Documentation

- [x] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
Make DDL unaffected by the KILL TiDB command"
```

### CODE REVIEW

**Comment by zimulala** [comment]
PTAL @tiancaiamao

File: sessionctx/variable/session.go [hunk 1]
```diff
@@ -75,6 +75,9 @@ var (
 	enableAdaptiveReplicaRead uint32 = 1
 )
 
+// ConnStatusShutdown indicates that the connection status is closed by server.
+const ConnStatusShutdown int32 = 2
```
Comment (lcwangchao) [inline sessionctx/variable/session.go:79]: Cloud you add some comments why this value must to be `2` to avoid modify it by mistake in the future?

**Review by lcwangchao — APPROVED** [review]

**Review by tiancaiamao — APPROVED** [review]

**Review by qw4990 — APPROVED** [review]

**Pull request merged**